pkg/controller.v1/pytorch/envvar.go

```diff
@@ -89,12 +112,26 @@ func setPodEnv(obj interface{}, podTemplateSpec *corev1.PodTemplateSpec, rtype,
 			// Set elastic related environment variables.
 			podTemplateSpec.Spec.Containers[i].Env = append(
 				podTemplateSpec.Spec.Containers[i].Env, envVars...)
+		} else {
+			podTemplateSpec.Spec.Containers[i].Env = append(
+				podTemplateSpec.Spec.Containers[i].Env, corev1.EnvVar{
+					Name:  EnvNnodes,
+					Value: strconv.Itoa(int(totalReplicas)),
+				})
 		}
 	}
 
 	return nil
 }
 
+func getNprocPerNode(job *kubeflowv1.PyTorchJob) int32 {
+	if job.Spec.NprocPerNode != nil {
+		return *job.Spec.NprocPerNode
+	} else {
+		return 1
+	}
+}
```

Comment:
Should we set the default value in [`Defaulter`](https://github.com/kubeflow/training-operator/blob/7e0eea6b2edbb55a3db35f3515cf28a2f2ccf7cf/pkg/apis/kubeflow.org/v1/pytorch_defaults.go) instead of here?

Reply:
Done